src/components/Carousel.tsx

```diff
@@ -1,18 +1,118 @@
 import React from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+type Props = {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinity: boolean,
+};
+
+type State = {
+  currentItemIndex: number,
+};
+
+export class Carousel extends React.Component <Props, State> {
+  state = {
+    currentItemIndex: 0,
+  };
+
+  handlerButton = (step: number) => () => {
```

Comment:
```suggestion
  handleButtonClick = (step: number) => () => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364